src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -679,19 +682,22 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		return undefined;
 	}
 
-	public mergeCells(cell: ICellModel, secondCell: ICellModel, newLinesRemoved: string[] | undefined): void {
-		let index = this._cells.findIndex(cell => cell.equals(cell));
-		if (index > -1) {
-			cell.source = newLinesRemoved.length > 0 ? [...cell.source, ...newLinesRemoved, ...secondCell.source] : [...cell.source, ...secondCell.source];
-			cell.isEditMode = true;
-			// Set newly created cell as active cell
-			this.updateActiveCell(cell);
-			this._contentChangedEmitter.fire({
-				changeType: NotebookChangeType.CellsModified,
-				cells: [cell],
-				cellIndex: index
-			});
-			this.deleteCell(secondCell, false);
+	public mergeCells(cells: ICellModel[], newLinesRemoved: string[] | undefined): void {
+		let firstCell = cells[0];
+		// Append the other cell sources to the first cell
+		for (let i = 1; i < cells.length; i++) {
+			firstCell.source = newLinesRemoved.length > 0 ? [...firstCell.source, ...newLinesRemoved, ...cells[i].source] : [...firstCell.source, ...cells[i].source];
```

Comment:
We don't want to add the newlines for every cell - the only one we should be adding this for is when adding on the tail cell